examples/exex/rollup/src/execution.rs

```diff
@@ -0,0 +1,489 @@
+fn construct_header(db: &Database, header: &RollupContract::BlockHeader) -> eyre::Result<Header> {
+    let parent_block = if !header.sequence.is_zero() {
+        db.get_block(header.sequence - U256::from(1))?
+    } else {
+        None
+    };
+
+    let block_number = u64::try_from(header.sequence)?;
+
+    // Calculate base fee per gas for EIP-1559 transactions
+    let base_fee_per_gas = if CHAIN_SPEC.fork(Hardfork::London).transitions_at_block(block_number) {
+        constants::EIP1559_INITIAL_BASE_FEE
+    } else {
+        parent_block
+            .as_ref()
+            .ok_or(eyre::eyre!("parent block not found"))?
+            .header
+            .next_block_base_fee(CHAIN_SPEC.base_fee_params_at_block(block_number))
+            .ok_or(eyre::eyre!("failed to calculate base fee"))?
+    };
+
+    // Construct header
+    Ok(Header {
+        parent_hash: parent_block.map(|block| block.header.hash()).unwrap_or_default(),
+        number: block_number,
+        gas_limit: u64::try_from(header.gasLimit)?,
+        timestamp: u64::try_from(header.confirmBy)?,
+        base_fee_per_gas: Some(base_fee_per_gas),
+        ..Default::default()
+    })
+}
+
+/// Configure EVM with the given database and header.
+fn configure_evm<'a>(
+    config: &'a EthEvmConfig,
+    db: &'a mut Database,
+    header: &Header,
+) -> Evm<'a, (), StateDBBox<'a, eyre::Report>> {
+    let mut evm = config.evm(
+        StateBuilder::new_with_database(Box::new(db) as DBBox<'_, eyre::Report>)
+            .with_bundle_update()
+            .build(),
+    );
+    evm.db_mut().set_state_clear_flag(
+        CHAIN_SPEC.fork(Hardfork::SpuriousDragon).active_at_block(header.number),
+    );
+
+    let mut cfg = CfgEnvWithHandlerCfg::new_with_spec_id(evm.cfg().clone(), evm.spec_id());
+    EthEvmConfig::fill_cfg_and_block_env(
+        &mut cfg,
+        evm.block_mut(),
+        &CHAIN_SPEC,
+        header,
+        U256::ZERO,
+    );
+    *evm.cfg_mut() = cfg.cfg_env;
+
+    evm
+}
+
+/// Decode transactions from the block data and recover senders.
+/// - If the transaction is a blob-carrying one, decode the blobs either using the local transaction
+///   pool, or querying Blobscan.
+/// - If the transaction is a regular one, decode the block data directly.
+async fn decode_transactions<Pool: TransactionPool>(
+    pool: &Pool,
+    tx: &TransactionSigned,
+    block_data: Bytes,
+    block_data_hash: B256,
+) -> eyre::Result<Vec<(TransactionSigned, Address)>> {
+    // Get raw transactions either from the blobs, or directly from the block data
+    let raw_transactions = if matches!(tx.tx_type(), TxType::Eip4844) {
+        let blobs: Vec<_> = if let Some(sidecar) = pool.get_blob(tx.hash)? {
+            // Try to get blobs from the transaction pool
+            sidecar.blobs.into_iter().zip(sidecar.commitments).collect()
+        } else {
+            // If transaction is not found in the pool, try to get blobs from Blobscan
```

Comment:
what is the right api for pulling blobs from CL? Should we open issue for this?

Reply:
coming here #7443